frontend/src/components/login/Login.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import './Login.css';
+import { LoginActions } from '../../actions/LoginActions';
+
+
+class Login extends Component{
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUsername = this.handleUsername.bind(this);
+    this.handlePassword = this.handlePassword.bind(this);
+    // this.login = this.login(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    let userData = this.state;
+    const { history } = this.props;
+
+    console.log(userData);
+    this.props.login(userData).then(success => {
+      if (success){
+        history.push('/kingdom');
+      }
+    });
+   }
+
+    handleUsername(e) {
+      let value = e.target.value;
+      this.setState(prevState => ({ ...prevState, username: value }));
+    }
+
+    handlePassword(e) {
+      let value = e.target.value;
+      this.setState(prevState => ({ ...prevState, password: value }));
```

Comment:
@zvorsi unnecessary value variable
```suggestion
      this.setState(prevState => ({ ...prevState, password: e.target.value }));
```

Reply:
tried, but immediately fails when trying to login, I'd rather leave it be